ddtrace/contrib/botocore/patch.py

```diff
@@ -328,16 +348,36 @@ def inject_trace_to_kinesis_stream_data(record, span):
         record["Data"] = data_json
 
 
-def inject_trace_to_kinesis_stream(params, span):
-    # type: (List[Any], Span) -> None
+def record_data_streams_path_for_kinesis_stream(pin, params, results):
+    stream_arn = params.get("StreamARN")
+
+    if not stream_arn:
+        log.debug("Unable to determine StreamARN for request with params: ", params)
+        return
+
+    processor = pin.tracer.data_streams_processor
+    for record in results.get("Records", []):
+        time_estimate = record.get("ApproximateArrivalTimestamp", datetime.now()).timestamp()
+        ctx = DataStreamsCtx(processor, 0, time_estimate, time_estimate)
```

Comment:
I don't think this will work, since set_checkpoint won't set parent_hash to 0 here. I will make a PR to be able to pass time_estimate directly to `set_checkpoint`

Reply:
Oh! I see you made that change actually, so here, no need to create this context right?
You can call `set_checkpoint` with time_estimate directly?

Reply:
I added in the capability to `set_checkpoint` but its not really needed.

What I believe my code does:

1. Create CTX with the arrival timestamp ([link](https://github.com/DataDog/dd-trace-py/blob/1.x/ddtrace/internal/datastreams/processor.py#L293)) which sets the "current edge start" to the estimated arrival and current hash to "0"
2.  Call set checkpoint which:
  a. skips [this section](https://github.com/DataDog/dd-trace-py/blob/1.x/ddtrace/internal/datastreams/processor.py#L343-L351)
  b. sets parent hash to 0 ([link](https://github.com/DataDog/dd-trace-py/blob/1.x/ddtrace/internal/datastreams/processor.py#L343-L351))
  c. computes path latencies using the values passed to create the DataStreamsCtx and time.time() ([link](https://github.com/DataDog/dd-trace-py/blob/1.x/ddtrace/internal/datastreams/processor.py#L358-L359))